libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -448,6 +448,32 @@ void AP_DDS_Client::on_topic(uxrSession* uxr_session, uxrObjectId object_id, uin
             GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Received sensor_msgs/Joy: %f, %f, %f, %f",
                           rx_joy_topic.axes[0], rx_joy_topic.axes[1], rx_joy_topic.axes[2], rx_joy_topic.axes[3]);
             // TODO implement joystick RC control to AP
+                    const uint32_t tnow = AP_HAL::millis();
+
+            //! @todo - to use same convention as MAVLink spec?
+            // see: void GCS_MAVLINK::handle_rc_channels_override
+
+            for (uint8_t i=0; i<MIN(8U, rx_joy_topic.axes_size); i++) {
```

Comment:
Why limit to 8 channels max?

Reply:
Adapted from code in `GCS_Common.cpp` : `GCS_MAVLINK::handle_rc_channels_override` where the treatment for the first 8 channels is slightly different.